### PR TITLE
Adding Archive Functionality

### PR DESCRIPTION
**Pull request by AbhinavThukral97**
This PR allows Drafts, Sessions, and Templates to be archived and creates a section for archived items for other actions.

### CODE REVIEW

**Comment by AbhinavThukral97** [comment]
WIP

**Comment by AbhinavThukral97** [comment]
This PR also contains changes from the Graph Templates PR as we wanted to add the archiving functionality to templates as well.

**Comment by houshuang** [comment]
Merge conflicts

**Comment by lfaucon** [comment]
@AbhinavThukral97 After merging there is a problem with a double definition of the function `parseTemplateData` in the file `imports/client/UserDashboard/data-utils/helpers.js`. Can you fix it ?

File: frog/imports/client/TeacherView/index.js [hunk 1]
```diff
@@ -14,7 +14,6 @@ import { Sessions } from '/imports/api/sessions';
 const TeacherViewRunner = withRouter(
   withTracker(({ match, history }) => {
     const user = Meteor.user();
-    let session;
```
Comment (lfaucon) [inline frog/imports/client/TeacherView/index.js]: Why this change? Doesn't it create syntax errors?

File: frog/imports/client/UserDashboard/components/ArchivesPage/index.js [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                  );
+                }
+              )}
+            </List>
+          </Paper>
+        </Grid>
+      )}
+    </Grid>
+  );
+};
```
Comment (lfaucon) [inline frog/imports/client/UserDashboard/components/ArchivesPage/index.js]: I don't like this file much. It seems to me that it is a near identical copy-paste of the `RecentPage` component. For code quality we should avoid such things. Do you see a way to reuse the same component instead of duplicating the code?

**Review by lfaucon — CHANGES_REQUESTED** [review]
Seems like a great feature to add. Thanks for this work!

Looks like this PR contains all the commits of #1883 and additional commits. I suggest merging #1883 first for this reason.

File: frog/imports/client/UserDashboard/components/ArchivesPage/index.js (hunk 2 above)
Comment (AbhinavThukral97) [reply frog/imports/client/UserDashboard/components/ArchivesPage/index.js]: The only difference between this and the `RecentPage` component is the display limit of 6 data rows. We can definitely add a variant in the RecentPage to accommodate this but my reasoning for keeping the two components separate was the future scope of expanding the RecentPage to add more functionality.

**Review by lfaucon — APPROVED** [review]
